src/core/toga/app.py

```diff
@@ -71,7 +71,7 @@ def __init__(self, name, app_id,
 
         self.default_icon = Icon('tiberius', system=True)
         self.icon = icon
-        self._main_window = None
+        self._windows = {'main': None}
```

Comment:
I can see what you're aiming at here; but there's a few problems with the "dictionary of human provided string" approach.

1. If a user provides the same window name twice, a window will be lost
2. If a user provides the window name of `"main"`, the main window will be lost (same problem, worse consequences)
3. It requires an "add and remove" API.

I'd rather see this done keyed by the window ID (which is a unique value, rather than a human provided one); and be based on assignment to the app. 

The current API means the main window is associated with the app by setting it as an attribute of the app:
```
app.main_window = some_window
```
This allows the app to track the window, and the window to know it's app. It should be possible to do the same with "other" windows by making the `windows` attribute expose an object that defines an `add` method:
```
app.windows.add(other_window)
```
`app.windows` can expose a set-like API (enabling iteration and add), but during the `add` call, you can do any additional tracking that is required (such as setting the app on the window itself).

Does that make sense?